lib/api/unittest/CPersistenceManagerTest.cc

```diff
@@ -287,8 +293,104 @@ class CTestFixture {
+        ml::api::CSingleStreamDataAdder dataAdder{dataStreamPtr};
+
+        // The 30000 second persist interval is set large enough that the timer
+        // will not trigger during the test - we bypass the timer in this test
+        // and kick off the background persistence chain explicitly
+        ml::api::CPersistenceManager persistenceManager{30000, false, dataAdder};
+
+        std::string description_;
+        std::string snapshotId_;
+        std::size_t numDocs_{0};
+
+        std::string backgroundSnapshotId;
+        std::string foregroundSnapshotId;
+
+        {
+            ml::core::CJsonOutputStreamWrapper wrappedOutputStream{outputStrm};
+
+            CTestAnomalyJob job{JOB_ID,
+                                limits,
+                                fieldConfig,
+                                modelConfig,
+                                wrappedOutputStream,
+                                std::bind(&reportPersistComplete,
+                                          std::placeholders::_1, std::ref(description_),
+                                          std::ref(snapshotId_), std::ref(numDocs_)),
+                                &persistenceManager,
+                                -1,
+                                "time",
+                                "%d/%b/%Y:%T %z"};
+
+            ml::api::CDataProcessor* firstProcessor{&job};
+
+            ml::api::CNdJsonInputParser parser{
+                {CTestFieldDataCategorizer::MLCATEGORY_NAME}, inputStrm};
+
+            BOOST_TEST_REQUIRE(parser.readStreamIntoMaps(
+                [firstProcessor](const ml::api::CDataProcessor::TStrStrUMap& dataRowFields) {
+                    return firstProcessor->handleRecord(
+                        dataRowFields, ml::api::CDataProcessor::TOptionalTime{});
+                }));
+
+            // Ensure the model size stats are up to date
+            job.finalise();
+
+            ml::core_t::TTime snapshotTimestamp{1283524206};
+            const std::string snapshotId{ml::core::CStringUtils::typeToString(snapshotTimestamp)};
+            const std::string description{"Supplied description for snapshot at " +
+                                          ml::core::CTimeUtils::toIso8601(snapshotTimestamp)};
+            BOOST_TEST_REQUIRE(job.doPersistStateInForeground(
+                dataAdder, description, snapshotId, snapshotTimestamp));
+
+            // Check that the snapshot description and Id reported by the "persist complete"
+            // handler match those supplied to the persist function
+            BOOST_REQUIRE_EQUAL(description, description_);
+            BOOST_REQUIRE_EQUAL(snapshotId, snapshotId_);
+
+            std::string state{dataStream->str()};
+
+            // Compare snapshot ID embedded in the state string with the supplied value.
+            const std::string expectedId{"job_model_state_" +
+                                         std::to_string(snapshotTimestamp) + "#1"};
```

Comment:
```suggestion
                                         snapshotId + "#1"};
```

It's just coincidence that the snapshot ID is the snapshot timestamp in this case.  The document ID is supposed to include the snapshot ID regardless of whether it's the same as the snapshot timestamp.